packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -28,31 +29,39 @@
    * <p>If no resizing is needed, returns the path for the original image.
    */
   String resizeImageIfNeeded(
-      String imagePath, Double maxWidth, Double maxHeight, int imageQuality) {
+      String imagePath, Double maxWidth, Double maxHeight, Integer imageQuality) {
     boolean shouldScale =
-        maxWidth != null || maxHeight != null || (imageQuality > -1 && imageQuality < 101);
+        maxWidth != null
+            || maxHeight != null
+            || (imageQuality != null && imageQuality > -1 && imageQuality < 101);
 
     if (!shouldScale) {
       return imagePath;
     }
 
     try {
-      File scaledImage = resizedImage(imagePath, maxWidth, maxHeight, imageQuality);
-      exifDataCopier.copyExif(imagePath, scaledImage.getPath());
+      Bitmap bmp = decodeFile(imagePath);
+      if (bmp == null) {
+        return null;
+      }
+      String[] pathParts = imagePath.split("/");
+      String imageName = pathParts[pathParts.length - 1];
```

Comment:
Are all images always guaranteed to have an extension?

Reply:
This is the imageName. For example if the imagePath is foo/bar/image.png, the `imageName` would be image.png.
Or you are asking what we should do if the `imageName` doesn't contain an extension?